core/products/future_test.go

```diff
@@ -36,47 +36,121 @@ func TestScalingOfSettlementData(t *testing.T) {
 }
 
 func testNoScalingNeeded(t *testing.T) {
-	ft := testFuture(t)
+	// Create test future with settlement data type integer with decimals (that represents a decimal)
+	ft := testFuture(t, datapb.PropertyKey_TYPE_INTEGER)
 
+	n := &num.Numeric{}
+	n.SetUint(num.NewUint(100000))
 	// settlement data is in 5 decimal places, asset in 5 decimal places => no scaling
-	scaled, err := ft.future.ScaleSettlementDataToDecimalPlaces(num.NewUint(100000), 5)
+	scaled, err := ft.future.ScaleSettlementDataToDecimalPlaces(
+		n, 5,
+	)
 	require.NoError(t, err)
 	require.Equal(t, num.NewUint(100000), scaled)
+
+	// Create test future with settlement data type decimal with decimals (that represents a decimal)
+	ft = testFuture(t, datapb.PropertyKey_TYPE_DECIMAL)
+
+	// settlement data is in 5 decimal places, asset in 3 decimal places => x10^-2
```

Comment:
comment doesn't apply anymore. Perhaps we should return an error if the settlement data has more decimals than the asset supports (ie settlement data in ETH can never be valid if it has more than 18 decimal places)

Reply:
Okay, this is what I missed. I get it.
I'll correct the formula now and the tests.
Thanks!